src/client/components/StoryFooter/Buttons.js

```diff
@@ -34,79 +37,100 @@ export default class Buttons extends React.Component {
     onEditClick: () => {},
   };
 
-  state = {
-    shareModalVisible: false,
-    shareModalLoading: false,
-    reactionsModalVisible: false,
-    loadingEdit: false,
-  };
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      shareModalVisible: false,
+      shareModalLoading: false,
+      reactionsModalVisible: false,
+      loadingEdit: false,
+    };
+
+    this.handleLikeClick = this.handleLikeClick.bind(this);
+    this.shareClick = this.shareClick.bind(this);
+    this.handleShareClick = this.handleShareClick.bind(this);
+    this.handleShareOk = this.handleShareOk.bind(this);
+    this.handleShareCancel = this.handleShareCancel.bind(this);
+    this.handleShowReactions = this.handleShowReactions.bind(this);
+    this.handleCloseReactions = this.handleCloseReactions.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+  }
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.postState.isReblogging !== this.props.postState.isReblogging) {
       this.setState({
         shareModalLoading: nextProps.postState.isReblogging,
-        shareModalVisible:
-          !(!nextProps.postState.isReblogging && this.props.postState.isReblogging) &&
-          this.state.shareModalVisible,
+        shareModalVisible: !(!nextProps.postState.isReblogging &&
+          this.props.postState.isReblogging) && this.state.shareModalVisible,
       });
     }
   }
 
-  handleShareClick = (e) => {
-    e.preventDefault();
+  handleLikeClick() {
+    this.props.onActionInitiated(this.props.onLikeClick);
+  }
+
+  shareClick() {
     if (this.props.postState.isReblogged) {
       return;
     }
 
     this.setState({
       shareModalVisible: true,
     });
-  };
+  }
 
-  handleShareOk = () => {
+  handleShareClick(e) {
+    e.preventDefault();
+    this.props.onActionInitiated(this.shareClick);
+  }
+
+  handleShareOk() {
     this.props.onShareClick();
-  };
+  }
 
-  handleShareCancel = () => {
+  handleShareCancel() {
     this.setState({
       shareModalVisible: false,
     });
-  };
+  }
 
-  handleShowReactions = () =>
+  handleShowReactions() {
     this.setState({
       reactionsModalVisible: true,
     });
+  }
 
-  handleCloseReactions = () =>
+  handleCloseReactions() {
     this.setState({
       reactionsModalVisible: false,
     });
+  }
 
-  handleCommentClick = () => {
+  // eslint-disable-next-line class-methods-use-this
+  handleCommentClick() {
```

Comment:
I would rather make it static method than disable the rule.

Reply:
sure we can make it a static method